apps/io-wallet-user-func/src/email-notification-service.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  MailerConfig,
+  getMailerTransporter,
+  sendMail,
+} from "@pagopa/io-functions-commons/dist/src/mailer";
+import * as TE from "fp-ts/lib/TaskEither";
+import * as t from "io-ts";
+
+import { MailConfig } from "./app/config";
+
+export const SendEmailParams = t.type({
+  html: t.string,
+  subject: t.string,
+  text: t.string,
+  to: t.string,
+});
+
+export type SendEmailNotificationParams = t.TypeOf<typeof SendEmailParams>;
+
+export interface IEmailNotificationService {
+  getTransporter: (
+    transporter: MailerConfig,
+    params: SendEmailNotificationParams,
+  ) => MailerConfig;
+  sendEmail: (
+    transporter: MailerConfig,
+    params: SendEmailNotificationParams,
+  ) => TE.TaskEither<Error, void>;
+}
+
+export class EmailNotificationService implements IEmailNotificationService {
+  #configuration: MailConfig;
+
```

Comment:
The class that implements a NotificationService tied to a particular infrastructure (MailUp) must not be implemented here but have the implementation inside the infra folder and make this file implementation agnostic. Then in the main file you can go and call the structure tied to your infrastructure.